marshmallow/fields.py

```diff
@@ -1311,6 +1314,20 @@ def _deserialize(self, value, attr, data, **kwargs):
 
         return result
 
+class Dict(Map):
+    """A dict field. Extends Map field with dict as the map_type.
+
+    Example: ::
+
+        numbers = fields.Map(keys=fields.Str(), values=fields.Float())
+
+    :param kwargs: The same keyword arguments that :class:`Map` receives.
+
+    .. versionadded:: 2.1.0
+    """
+    def __init__(self, **kwargs):
+        super(Dict, self).__init__(map_type=dict, **kwargs)
```

Comment:
You could reduce the boilerplate by making this declarative and allowing the `Mapping` field to default to the `Mapping` collection type.

```py
class Dict(Mapping):
    mapping_type=collections.Mapping
    ...

class Dict(Mapping):
    mapping_type=dict
```

Reply:
Issue with this is if I want to create my own custom dict, I need to create class and change field my inheriting it. If we pass the mapping type as argument, it will be as easy as `myfield = fields.Mapping(map_type=MyCustomDict)`

Reply:
Actually I was thinking from the user's perspective i.e. how easily an user can define a field in their schema.
In case of Dict, user doesn't need to pass any extra argument... only keys, values and kwargs. User need to pass map_type only while initializing Mapping field.

If we take the alternate approach, we need to define Mapping as abstract class. Check #1098

Reply:
Implementing a custom class in userland is the way to go, and consistent with how `Number` is implemented. OK, let's continue on #1098.